torchdata/dataloader2/graph/settings.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+import inspect
+
+from typing import Optional
+
+import torch
+
+from torchdata.dataloader2.graph import DataPipe, find_dps, list_dps, traverse_dps
+from torchdata.datapipes.iter import ShardingFilter
+
+
+def _is_random_datapipe(datapipe: DataPipe) -> bool:
+    if hasattr(datapipe, "set_seed") and inspect.ismethod(datapipe.set_seed):
+        return True
+    return False
+
+
+def _generate_random_seed(rng: Optional[torch.Generator] = None, dtype: torch.dtype = torch.int64) -> int:
+    return int(torch.empty((), dtype=dtype).random_(generator=rng).item())
+
+
+def _seed_datapipe(datapipe: DataPipe, seed_generator: torch.Generator) -> DataPipe:
+    seed = _generate_random_seed(seed_generator)
+    datapipe.set_seed(seed)
+    return datapipe
+
+
+def _set_graph_seeds(datapipe: DataPipe, seed_generator: torch.Generator, worker_id: int = 0) -> DataPipe:
+    r"""
+    Set seeds to the graph of ``DataPipes`` based on a Seed Generator. All random ``DataPipes`` prior to
+    ``ShardingFilter`` will be set seeds by the same Seed Generator to preserve the same random state
+    across distributed/non-distributed workers. And, the random ``DataPipes`` after ``ShardingFilter``
+    will be set seeds by the worker-local Seed Generator deterministically created based on ``worker_id``.
+    """
```

Comment:
Add a note about:
1. `worker_seed = base_seed + worker_id`, currently no option to override
2. If `ShardingFilter` is not used, shuffle seeds (and any other RNG) in workers will be different.
3. Any discrepancy between DL1 vs DL2 (might need randomness test??)

Reply:
I will add those comments to the ReadingService in the next PR. This function is not exposed to users for now and no online doc for it as well.